src/scripting/vmthunks.cpp

```diff
@@ -2503,12 +2504,146 @@ DEFINE_ACTION_FUNCTION(_Console, MidPrint)
 	return 0;
 }
 
+//==========================================================================
+//
+//
+//
+//==========================================================================
+
+static int isValid( level_info_t *info )
+{
+	return info->isValid();
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, isValid, isValid)
+{
+	PARAM_SELF_STRUCT_PROLOGUE(level_info_t);
+	ACTION_RETURN_BOOL(self->isValid());
+}
+
+static void ZLookupLevelName( level_info_t *info, FString *result )
+{
+	*result = info->LookupLevelName();
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, LookupLevelName, ZLookupLevelName)
+{
+	PARAM_SELF_STRUCT_PROLOGUE(level_info_t);
+	ACTION_RETURN_STRING(self->LookupLevelName());
+}
+
+static int GetLevelInfoCount()
+{
+	return wadlevelinfos.Size();
+}
+
+static level_info_t* GetLevelInfo( int index )
+{
+	return &wadlevelinfos[index];
```

Comment:
Please add bounds checking. Also, this function should be placed right before its  `DEFINE_ACTION_FUNCTION_NATIVE` to match other definitions.